tests/unittests/test_salesforce_null_bookmark.py

```diff
@@ -0,0 +1,39 @@
+import logging
+from typing import Counter
+import unittest
+from unittest import mock
+# from tap_salesforce import main
+import tap_salesforce
+from tap_salesforce import Salesforce, metrics
+from tap_salesforce.sync import sync_records
+import json
+
+
+
+class TestNullBookmarkTesting(unittest.TestCase):
+    @mock.patch('tap_salesforce.salesforce.Salesforce.query', side_effect=lambda test1, test2: [])
+    def test_not_null_bookmark_for_incremental_stream(self, mocked_query):
+        """
+        To ensure that after resolving writebook mark logic not get "Null" key in state
```

Comment:
Add detailed comment regarding what should be null in the state and which key-value pair is not getting emitted now.

Reply:
Added detailed comment for test function